spec/controllers/prompt_meme_controller_spec.rb

```diff
@@ -3,6 +3,46 @@
 describe Challenge::PromptMemeController do
   include LoginMacros
 
+  describe "new" do
+    context "when the collection already has a challenge" do
+      before do
+        @collection = FactoryGirl.create(:collection, challenge: PromptMeme.new)
+        fake_login_known_user(@collection.owners.first.user)
+        post :new, params: { collection_id: @collection.name }
+      end
+
+      it "should set flash notice" do
```

Comment:
Because it's something we test a lot, we've set up a bunch of [redirect expectation helpers](https://github.com/otwcode/otwarchive/blob/master/spec/support/redirect_expectation_helper.rb) that we prefer to use for testing redirects with notice or error messages. For example, we might condense this down to:

```
context "when the collection already has a challenge" do
  [...]

  it "redirects to challenge edit page with notice"
    it_redirects_to_with_notice(edit_collection_prompt_meme_path(@collection), "There is already a challenge set up for this collection.")
  end
```

Generally, we prefer to leave off the "should" in our descriptions, per [these guidelines](http://www.betterspecs.org/#should).